webapp/security/views.py

```diff
@@ -333,5 +342,182 @@ def cve(cve_id):
 # ===
 
 
-def api_create_cve():
-    pass
+def create_cve(cve_id):
+    """
+    Receives a POST request from load_cve.py
+    Parses the object and bulk inserts with add_all()
+    @params json: the body of the request
+    """
+
+    data = flask.request.json
+    packages = []
+    references = []
+    bugs = []
+
+    # Check if there are packages before mapping
+    if data.get("packages"):
+        for package_data in data["packages"]:
+            releases = []
+            for rel in package_data["releases"]:
+                releases.append(
+                    PackageReleaseStatus(
+                        name=rel["name"],
+                        status=rel["status"],
+                        status_description=rel["status_description"],
+                    )
+                )
+            packages.append(
+                Package(
+                    name=package_data["name"],
+                    source=package_data["source"],
+                    ubuntu=package_data["ubuntu"],
+                    debian=package_data["debian"],
+                    releases=releases,
+                )
+            )
+
+    if data.get("references"):
+        for ref in data["references"]:
+            references.append(CVEReference(uri=ref))
+
+    if data.get("bugs"):
+        for b in data["bugs"]:
+            bugs.append(Bug(uri=b))
+
+    cves = CVE(
+        id=data["id"],
+        status=data.get("status", ""),
+        last_updated_date=data.get("last_updated_date"),
+        public_date_usn=data.get("public_date_usn"),
+        priority=data.get("priority"),
+        cvss=data.get("cvss"),
+        assigned_to=data.get("assigned_to"),
+        discovered_by=data.get("discovered_by"),
+        approved_by=data.get("approved_by"),
+        description=data.get("description"),
+        ubuntu_description=data.get("ubuntu_description"),
+        notes=data.get("notes", []),
+        packages=packages,
+        references=references,
+        bugs=bugs,
+    )
+
+    try:
+        db_session.add(cves)
+        db_session.commit()
+    # Duplicate key errors
+    except exc.IntegrityError as error:
+        return flask.jsonify({"message": error.orig.args[0]}), 409
+    # Payload input errors
+    except exc.DataError as error:
+        return (
+            flask.jsonify({"message": error.orig.args[0]}),
+            400,
+        )
+    except exc.SQLAlchemyError as error:
+        return flask.jsonify({"message": error}), 400
```

Comment:
What cases couldn't be successfully handled by a `DBAPIError`?

```suggestion
    except exc.DBAPIError as error:
        return flask.jsonify({"message": error.orig.args[0]}), 400
```

Reply:
Yes, these catch duplicate key error (CVE id that already exists) and incorrect input format (in this case only datetime)